### PR TITLE
Add policy links command

### LINKED ISSUE #1873: [policy] mc policy links s3/mybucket to list fully qualified URLs
This was requested by @atomi to list public URLs for external processing.

### PR DESCRIPTION
**Pull request by vadmeste**
Fixes #1873

### CODE REVIEW

**Comment by vadmeste** [comment]
This PR is ready for review, the only existing problem is that it accepts --recursive and --json flag when they are before 'links' command and not after, let's discuss other issues if found first.

File: cmd/policy-main.go [hunk 1]
```diff
@@ -206,72 +255,151 @@ func doGetAccessRules(targetURL string) (r map[string]string, err *probe.Error)
...
+		for content := range clnt.List(isRecursive, isIncomplete, DirFirst) {
+			if content.Err != nil {
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list folder.")
+				continue
+			}
+			// Construct the message to be displayed to the user
+			msg := policyLinksMessage{}
+			msg.Status = "success"
+			// Convert any os specific delimiters to "/".
+			msg.FullURL = filepath.ToSlash(content.URL.String())
```
Comment (harshavardhana) [inline cmd/policy-main.go]: Don't do 'ToSlash' for URLs on windows the output will have '\' which is unexpected.

File: cmd/policy-main.go [hunk 2]
```diff
@@ -206,72 +255,151 @@ func doGetAccessRules(targetURL string) (r map[string]string, err *probe.Error)
...
+			if content.Type.IsDir() {
+				msg.Type = "folder"
```
Comment (harshavardhana) [inline cmd/policy-main.go]: We can skip the 'CommonPrefix' since that is not useful in downloading the object itself.

Comment (harshavardhana) [inline cmd/policy-main.go]: We can remove the Type itself it just prints fully qualified object URLs.

**Comment by vadmeste** [comment]
All comments are addressed

**Pull request merged**